### PR TITLE
soa test & debug 

### LINKED ISSUE #75: [SOA bot] Send debugging messages related to stance vector to GLOBAL channel

### LINKED ISSUE #89: [SOA bot] Generate support proposals

### PR DESCRIPTION
**Pull request by wwongkamjan**
this is to update current version of SOA_test (also pulled current version of SOA) 

Also, to @kartik2112 to add new test function responding to

```
respond_to_invalid_orders,
respond_to_alliance_messages
HUH 
ALY
```

Resolves #89 
Resolves #75

### CODE REVIEW

**Comment by wwongkamjan** [comment]
ATM, I have done testing

```
        self.test_play
        self.test_stance
        self.test_parse_proposals
```

**Comment by trigaten** [comment]
CI is failing due to introduction of tornado into gameplay framework. Since SOAbot will use the TACC http server, it would be better to spin up an instance of his for the bot to be tested on.

**Comment by kartik2112** [comment]
Resolves #75

**Comment by kartik2112** [comment]
Commented the soa functions invocation. Will wait for Feng to complete 1st version of tests through CI for dipnet related bots.

**Comment by kartik2112** [comment]
Resolves #89

File: .gitignore [hunk 1]
```diff
@@ -11,3 +11,6 @@ build
 .vscode
 ./joblogs/*
 ./jobs/*
+*/batch.txt
+**/bot_neurips2019-sl_model
```
Comment (trigaten) [inline .gitignore:15]: What do */ and **/ do here?

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -25,5 +25,6 @@ jobs:
     - name: "Run tests"
       run: |
         pytest tests/utils_test.py
-        pytest tests/bot_tests/bots_test.py
+        # pytest tests/bot_tests/bots_test.py
```
Comment (trigaten) [inline .github/workflows/CI.yml:28]: why?

Comment (trigaten) [inline .github/workflows/CI.yml:28]: why is this commented out?

Comment (trigaten) [reply .github/workflows/CI.yml:28]: Why would these tests be failing?

File: tests/bot_tests/gameplay_framework.py [hunk 3]
```diff
@@ -2,11 +2,16 @@
...
 
 from typing import List
-
-from diplomacy import Game, Message
+from tornado import gen
+from diplomacy import Game, Message, connect
 from diplomacy.utils.export import to_saved_game_format
 
 from baseline_bots.bots.baseline_bot import BaselineBot, BaselineMsgRoundBot
+from baseline_bots.bots.random_proposer_bot import RandomProposerBot_AsyncBot
```
Comment (trigaten) [inline tests/bot_tests/gameplay_framework.py:10]: why import these bots?

File: tests/bot_tests/gameplay_framework.py [hunk 4]
```diff
@@ -96,27 +104,23 @@ def step(self):
...
         return {"messages": msgs_to_send}, self.game.is_game_done
             
-            
-
-            
-        
 
+@gen.coroutine            
+def game_loop():
+    game_play_obj = GamePlay(None, [RandomProposerBot_AsyncBot, RandomProposerBot_AsyncBot, RandomProposerBot_AsyncBot, SmartOrderAccepterBot, SmartOrderAccepterBot, SmartOrderAccepterBot, SmartOrderAccepterBot], 3, True)
```
Comment (trigaten) [inline tests/bot_tests/gameplay_framework.py]: this specific test should not be in this file

Comment (wwongkamjan) [reply tests/bot_tests/gameplay_framework.py]: I think this helps those students that have to test-run SOA for the game loop with their new function. They don't need to go to SOA_test.py just yet.

**Review by wwongkamjan — COMMENTED** [review]
let's merge this branch before it diverges even more

**Pull request merged**